client/analysisui/rmanovafactorscontrol.js

```diff
@@ -60,13 +63,83 @@ var rmafcItem = function(parent, data, isFirst, isLast) {
 
         if (isEmpty === false) {
             for (var i = 0; i <= levels.length; i++)
-                this.createLevel(levels[i], i);
+                this.createLevel(index, levels[i], i);
+        }
+    };
+
+    this.intToRoman = function(number) {
```

Comment:
this is pretty funny that you've written a function which assembles roman numerals up to 4000, but couldn't this be written more easily like this:

```
function intToRoman(number)
    let text;
    text  = [ '', 'I', 'II', ...][parseInt(number) % 10]
    text += [ '', 'X', 'XX', ...][parseInt(number / 10) % 10]
    text += [ '', 'C', 'CC', ...][parseInt(number / 100) % 10]
    text += [ '', 'M', 'MM', ...][parseInt(number / 1000) % 10]
    return text;
}
```